src/resources/env_config.rs

```diff
@@ -0,0 +1,233 @@
+    }
+}
+
+/// Configuration information for all the third-party resources. This specifies
+/// for every resource whether it should be included at all, and if so, just as
+/// an URL or also as an embedded resource.
+pub struct Configuration {
+    pub jquery: BasicResource,
+    pub mathjax: BasicResource,
+    pub highlight: BasicResource,
+    pub awesome: Awesome,
+    pub open_sans: BasicResource,
+    pub source_code_pro: BasicResource
+}
+
+/// Calculate an environment variable name of the format MDBOOK_[name]_[key]
+fn varname(resource : &str, key: &str) -> String {
+    "MDBOOK_".to_string() + &resource.to_uppercase() + "_" + key
+}
+
+/// Returns the contents of the environment variable of the given resource with
+/// the given key, or None if the environment variable was not set.
+fn var(resource: &str, key: &str) -> Option<String> {
+    env::var(varname(resource, key)).ok()
+}
+
+/// Returns a variable as var would. If the variable could not be found, return
+/// the given default.
+fn var_default(resource: &str, key: &str, default: &str) -> String {
+    var(resource, key).unwrap_or(String::from(default))
+}
+
+/// The rendering strategy for a resource
+enum Strategy {
+    /// Render both an inclusion through an URL, and a fallback if it exists.
+    UrlWithFallback,
+    /// Render only the inclusion through an URL
+    UrlOnly,
+    /// Omit the resource altogether
+    Omit
+}
+
+/// Return a strategy from an environment variable, or the default
+/// `UrlWithFallback` if it can't be found.
+fn strategy(resource: &str) -> Strategy {
+    match var(resource, "STRATEGY")
+        .unwrap_or(var_default("DEFAULT", "STRATEGY", "UrlWithFallback")).as_ref() {
+            "UrlOnly" => Strategy::UrlOnly,
+            "Omit" => Strategy::Omit,
+            _ => Strategy::UrlWithFallback
+    }
+}
+
+/// return an URL from an environment variable, or the given default if the URL
+/// was not set.
+fn url(resource: &str, default: &str) -> String {
+    var_default(resource, "URL", default)
+}
+
+/// return a source location from an environment variable, if it can't be found.
+fn source(resource : &str) -> Option<String> {
+    var(resource, "EMBED_SOURCE")
+}
+
+/// return a resource from various environment variables.
+fn resource(resource: &str, url_default: &str) -> BasicResource {
+    match strategy(resource) {
+        Strategy::UrlWithFallback => BasicResource::UrlWithFallback {
+            url: url(resource, url_default),
+            source: source(resource)
+        },
+        Strategy::UrlOnly => BasicResource::UrlOnly {
+            url: url(resource, url_default)
+        },
+        Strategy::Omit => BasicResource::Omit
+    }
+}
+
+/// parse font_awesome configuration from environment variables.
+fn awesome() -> Awesome {
+    Awesome { resource: resource("AWESOME", AWESOME_URL) }
+}
+
+/// Parse the configuration from the environment variables.
+pub fn configuration_from_env() -> Configuration {
```

Comment:
Can we add a couple integration tests (most probably in the `tests/` directory) which set some environment variables and make sure `Configuration_from_env()` picks up the appropriate resources? The implementation itself is fairly straightforward, but it's always nice to have a couple sanity tests to make sure it all works and prevent any regressions from being introduced down the track.